adal-analytics/src/main/java/com/massivedisaster/adal/analytics/AnalyticsManager.java

```diff
@@ -178,4 +182,97 @@ public void sendEvent(int categoryId, int actionId, String label) {
 
         mTracker.send(builder.build());
     }
+
+    /**
+     * Send an product impression event
+     *
+     * @param product    the product
+     * @param label      the label
+     * @param screenName the screenName
+     */
+    public void sendImpression(Product product, String label, String screenName) {
+        if (getTracker() == null || product == null) {
+            return;
+        }
+
+        HitBuilders.ScreenViewBuilder builder = new HitBuilders.ScreenViewBuilder()
+                .addImpression(product, label);
+
+        mTracker.setScreenName(screenName);
+        mTracker.send(builder.build());
+    }
+
+    /**
+     * Send an product action event
+     *
+     * @param action     the action
+     * @param product    the product
+     * @param screenName the screenName
+     */
+    public void sendProduct(@NonNull ProductAction action, @Nullable Product product, String screenName) {
+        if (getTracker() == null) {
+            return;
+        }
+
+        HitBuilders.ScreenViewBuilder builder = new HitBuilders.ScreenViewBuilder();
+        if (product != null) {
+            builder.addProduct(product);
+        }
+
+        builder.setProductAction(action);
+
+        mTracker.setScreenName(screenName);
+        mTracker.send(builder.build());
+    }
+
+    /**
+     * Send an promotion impression event
+     *
+     * @param promotion  the promotion
+     * @param screenName the screenName
+     */
+    public void sendPromotion(@NonNull Promotion promotion, String screenName) {
+        if (getTracker() == null) {
+            return;
+        }
+
+        HitBuilders.ScreenViewBuilder builder = new HitBuilders.ScreenViewBuilder()
+                .addPromotion(promotion);
+
+        mTracker.setScreenName(screenName);
+        mTracker.send(builder.build());
+    }
+
+    /**
+     * Send an promotion action event
+     *
+     * @param promotion  the promotion
+     * @param screenName the screenName
+     */
+    public void sendPromotionAction(@NonNull Promotion promotion, @NonNull String promotionAction, @Nullable String category, @Nullable String action,
+                                    @Nullable String label, String screenName) {
+        if (getTracker() == null) {
+            return;
+        }
+
+        HitBuilders.EventBuilder builder = new HitBuilders.EventBuilder()
+                .addPromotion(promotion)
+                .setPromotionAction(promotionAction);
+
+        if (category != null) {
```

Comment:
didn't check the source code but this nullity validation is necessary? `builder.setCategory(parameter)` throws if the parameter is null?